### DISCUSSION TITLE
Questions about iteration

### DISCUSSION THREAD

**Discussion by dingjy0714** [kaust-halo/geeet#32] [Q&A]
When I used this geeet and the TSEB model to calculate the regional ET, I found the obtained results:  
1. Tc=Ts
2. the minimum value of LEc and LEs was negative.  
3. the minimum value of ETd was also negative.
Then I checked the code and found that you had set the maximum number of iterations to be 5. But after 5 iterations my results were not satisfactory. So I changed the code in tseb.py so that **max_iterations = 5**, which ensures that PT is positive, which ensures that alphaPT is positive. But when I run it again, the result is the same as before. How can I solve it ?
In addition, in this geeet, the condition to stop the iteration is that LEs is positive. Should we add a condition that L converges to a stable value ?

Looking forward to your reply !

![K43BLHHZ1DHMK8G6NI E)UW](https://github.com/kaust-halo/geeet/assets/163144863/b698667b-d662-4772-b7a8-63083d399331)

**Comment by lopezvoliver** [comment]
Hi, it looks like the screenshot shows the Asset details of a GEE image, and that you used an export scale of 1000 m. 

Would you mind sharing the asset so we can take a better look?

Note that the table shows the approximate min and max band values, so it's not surprising that there are negative values in the min column. 

Note that you don't need to change the code in `tseb.py`, as the `tseb_series` function accepts `max_iterations` as a keyword argument. 

For the convergence of L, if you have an improvement for the code based on this idea, you are encouraged to implement and submit a pull request to incorporate it into the code. 

I am moving this conversation to the discussion section as there is no specific issue to address here.

**Reply by dingjy0714** [reply]
I would be happy to share with you the details of my GEE assets. The link is: [https://code.earthengine.google.com/?asset=users/djy00714/geeet/LC08_132033_20200701](url)

In fact, these negative values bothered me for a long time. Thank you very much for your explanation.

**Reply by lopezvoliver** [reply]
You will need to click the "share" button on the asset, otherwise I can't read it.

**Reply by dingjy0714** [reply]
Sorry,  I forgot to check "Anyone can read". I was really careless.

Let’s try again: [https://code.earthengine.google.com/?asset=users/djy00714/geeet/LC08_132033_20200701](url)
